src/compat/libc/string/memccpy.c

```diff
@@ -0,0 +1,99 @@
+/**
+ *  @file
+ *  @brief Implementation of #memccpy() function.
+ *
+ *  @date 18/3/20
+ *  @author Giannis Zapantis (github name: tech-gian)
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+void* memccpy(void* dest, const void* src, int c, size_t n) {
+
+/* If overlap happens the results
+ * will be undefined.
+ */
+
+
+/* if there is already data after c
+ * then that data stays the same.
+ *
+ * if there isn't we don't change
+ * anything.
+ */
+
+/* copying dest to temp_dest
+ * and src to temp_dest.
+ *
+ * So we can work for each
+ * byte separately.
+*/
+
+    char* temp_dest = (char*) dest;
+    char* temp_src = (char*) src;
+    size_t pos = n;
+
+    char* ar1[n];
+
+/* getting each byte for c. */
+    char temp_c[4];
+    temp_c[0] = (c >> 24) & 0xFF;
+    temp_c[1] = (c >> 16) & 0xFF;
+    temp_c[2] = (c >> 8) & 0xFF;
+    temp_c[3] = c & 0xFF;
+
+
+/* if c!=0 then we check the last byte.
+ *
+ * if there is in src.
+ */
+    if (c!=0) {
+        for (size_t i = 0; i < n; i++) {
+            if (temp_src[i] == temp_c[3] && temp_c[3] != 0) {
+                pos = i;
+                break;
+            }
+        }
+    }
+    else {
+/* if c==0 we check every byte to be
+ * the same (4 bytes of int).
+ */
+        for (size_t i = 0; i < n; i++) {
+            if (i < n - 4) {
+                if (temp_src[i] == temp_c[0] && temp_src[i+1] == temp_c[1] && temp_src[i+2] == temp_c[2] && temp_src[i+3] == temp_c[3]) {

```

Comment:
what do you check here?

Reply:
If c==0 then we have to check all 4 bytes of c (because it's int), to see if it really equals with an element of src. Else we only have to check byte to byte to see if it is the same. The problem came up when I tested this function with ints and send 0 as c (stopping character).